GitUI/Script/IScriptRunner.cs

```diff
@@ -0,0 +1,14 @@
+﻿using GitCommands;
+
+namespace GitUI.Script
+{
+    internal interface IScriptRunner
+    {
+        /// <summary>
+        /// Tries to run scripts identified by a <paramref name="command"/>
+        /// </summary>
+        CommandStatus ExecuteScriptCommand(int command);
+
+        CommandStatus RunScript(string scriptKey);
```

Comment:
Maybe

RunByKey
RunByName

I understand correctly that there can not be two scripts with the same 'int command'?

public CommandStatus ExecuteScriptCommand(int command)
        {
            var anyScriptExecuted = false;
            var needsGridRefresh = false;

            foreach (var script in _scriptManager.GetScripts())
            {
                if (script.HotkeyCommandIdentifier == command)
                {
                    var result = RunScript(script.Name);

                    anyScriptExecuted = true;
                    needsGridRefresh |= result.NeedsGridRefresh;
                }
            }

            return new CommandStatus(anyScriptExecuted, needsGridRefresh);
        }

I think we can separate the search, validation and execution of the script

Reply:
I would go as far as
```
CommandStatus RunScript(int command);
CommandStatus RunScript(string scriptKey);
```

Reply:
> I understand correctly that there can not be two scripts with the same 'int command'?

In theory a situation like this should be prohibited. In reality I don't believe there is anything to prevent this now.

Ideally I'd like to have a central service that any component within the app can register with to say what event they can handle, and publish events to - basically a pubsub. 
This way we could have a register of events, events could be arbitrary strings rather than numbers.

Once upon the time I even did ticker with the code, however it looks like I've lost the branch... Though it would probably won't matter, the codebase has evolved since then.

If you wish and have capacity, let's try to build it for the next release 😉 
I'll do my best to help.

Reply:
I do not think that I fully understand how this should work. Perhaps there is some kind of scheme or sequence of actions that should be followed up?